Analysis/Tasks/PWGHF/qaTask.cxx

```diff
@@ -55,209 +32,130 @@ double ConvertPhiRange(double phi)
 /// \param impactParameterZ variable to save the impact parameter (in Z) in micrometers.
 /// \param impactParameterZError variable to save the impact parameter (in Z) error in micrometers.
 template <typename Track>
-bool GetImpactParameterAndError(const Track& track, const o2df::VertexBase& primaryVertex, double& impactParameterRPhi,
-                                double& impactParameterRPhiError, double& impactParameterZ,
+bool GetimpactParameterAndError(const Track& track,
+                                const VertexBase& primaryVertex,
+                                double& impactParameterRPhi,
+                                double& impactParameterRPhiError,
+                                double& impactParameterZ,
                                 double& impactParameterErrorZ)
 {
   impactParameterRPhi = -999.;
   impactParameterRPhiError = -999.;
-  impactParameterZ = -999;
-  impactParameterErrorZ = -999;
+  impactParameterZ = -999.;
+  impactParameterErrorZ = -999.;
 
-  o2df::DCA dca;
+  DCA dca;
   // FIXME: get this from CCDB
   constexpr float magneticField{5.0}; // in kG
   auto trackParameter = getTrackParCov(track);
-  bool propagate = trackParameter.propagateToDCA(primaryVertex, magneticField, &dca);
+  bool propagate = trackParameter.propagateToDCA(primaryVertex, magneticField, &dca, 100.);
 
-  constexpr float conversion_to_micrometer = 1000;
+  constexpr float toMicrometers = 10000.f; // Conversion from [cm] to [mum]
   if (propagate) {
-    impactParameterRPhi = conversion_to_micrometer * dca.getY();
-    impactParameterRPhiError = conversion_to_micrometer * std::sqrt(dca.getSigmaY2());
-    impactParameterZ = conversion_to_micrometer * dca.getZ();
-    impactParameterErrorZ = conversion_to_micrometer * std::sqrt(dca.getSigmaZ2());
+    impactParameterRPhi = toMicrometers * dca.getY();
+    impactParameterRPhiError = toMicrometers * std::sqrt(dca.getSigmaY2());
```

Comment:
If I am not missing anything, these are expressed in Cartesian _x_, _y_ coordinates, not cylindrical _r_, _φ_.

Reply:
I think Rphi is intended in the 3D space, it should be fine